From: Egbert Eich <eich@suse.de>
Date: Tue Apr 12 09:22:40 2016 +0200
Subject: [PATCH]Install: Avoid failure on wrapper installation
Patch-mainline: never
References: 
Signed-off-by: Egbert Eich <eich@suse.com>

- Check for SUID_WRAPPER_DIR being identical to bindir
  before copying script.
- Check whether user is root before doing a chmod/chown

Signed-off-by: Egbert Eich <eich@suse.de>
---
 hw/xfree86/Makefile.am | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

Index: xorg-server-1.20.5/hw/xfree86/Makefile.am
===================================================================
--- xorg-server-1.20.5.orig/hw/xfree86/Makefile.am
+++ xorg-server-1.20.5/hw/xfree86/Makefile.am
@@ -113,9 +113,10 @@ if INSTALL_SETUID
 endif
 if SUID_WRAPPER
 	$(MKDIR_P) $(DESTDIR)$(SUID_WRAPPER_DIR)
-	mv $(DESTDIR)$(bindir)/Xorg $(DESTDIR)$(SUID_WRAPPER_DIR)/Xorg
-	${INSTALL} -m 755 Xorg.sh $(DESTDIR)$(bindir)/Xorg
-	-chown 0 $(DESTDIR)$(SUID_WRAPPER_DIR)/Xorg.wrap && chmod u+s $(DESTDIR)$(SUID_WRAPPER_DIR)/Xorg.wrap
+	mv $(DESTDIR)$(bindir)/Xorg $(DESTDIR)$(SUID_WRAPPER_DIR)/Xorg 2>/dev/null && target=Xorg; \
+	${INSTALL} -m 755 Xorg.sh $(DESTDIR)$(bindir)/$${target}
+	-test "x$UID" = "x0" -o "x$EUID" = "x0" && \
+		chown root $(DESTDIR)$(SUID_WRAPPER_DIR)/Xorg.wrap && chmod u+s $(DESTDIR)$(SUID_WRAPPER_DIR)/Xorg.wrap
 endif
 
 uninstall-local:
